python/lsst/ts/externalscripts/make_love_stress_tests.py

```diff
@@ -0,0 +1,324 @@
+                    for name in self.telemetry_streams:
+                        csc, salindex = salobj.name_to_name_index(name)
+                        for stream in self.telemetry_streams[name]:
+                            await self.__subscribe_to(
+                                csc, salindex, stream, "telemetry"
+                            )
+                    await self.__handle_message_reception()
+        except Exception as e:
+            raise RuntimeError("Manager Client connection failed.") from e
+
+    async def close(self):
+        if self.__websocket:
+            await self.__websocket.close()
+
+
+class StressLOVE(salobj.BaseScript):
+    """Run a stress test for one or more CSCs.
+
+    Notes
+    -----
+    **Details**
+
+    * Run a LOVE stress test by generating several client connections
+    that will listen to every event and telemetry of the specified CSCs
+    """
+
+    def __init__(self, index):
+        super().__init__(index=index, descr="Run a stress test for one or more CSCs")
+
+        # SimpleNamespace to store stress test configurations
+        # params described on `get_schema`
+        self.config = None
+
+        # list to store clients connections,
+        # each one an instance of ManagerClient
+        self.clients = []
+
+        # dict to store remote connections,
+        # with each item in the form of
+        # `CSC_name[:index]`: `lsst.ts.salobj.Remote`
+        self.remotes = {}
+
+        # commands timeout
+        self.cmd_timeout = 10
+
+    @classmethod
+    def get_schema(cls):
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_standardscripts/StressLOVE.yaml
+            title: StressLOVE v1
+            description: Configuration for StressLOVE
+            type: object
+            properties:
+              location:
+                description: Domain of the running LOVE instance to stress
+                type: string
+              number_of_clients:
+                description: The number of clients to create
+                type: number
+              number_of_messages:
+                description: The number of messages to store before calculating the mean latency
+                type: number
+              data:
+                description: List of CSC_name[:index]
+                type: array
+                minItems: 1
+                items:
+                    type: string
+            required: [location, number_of_clients, number_of_messages, data]
+            additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    def set_metadata(self, metadata):
+        """Compute estimated duration.
+
+        Parameters
+        ----------
+        metadata : SAPY_Script.Script_logevent_metadataC
+        """
+        # a crude estimate;
+        metadata.duration = len(self.clients) * 5
```

Comment:
I think this will also depend more on `number_of_messages` then on the number of clients no? Maybe create a "expected_message_frequency" parameter to have the multiplying factor? So this would be:

```
metadata.duration = self.config.number_of_messages / self.expected_message_frequency
```

And you define `expected_message_frequency` in the `__init__` method.

Reply:
That's correct. I'll adjust this :+1: